File: 20-ValidParentheses/20-ValidParentheses.java
// Last updated: 7/12/2026, 6:25:43 PM
public class Solution {
public boolean isValid(String s) {
    char [] arr = s.toCharArray();
	Stack stack = new Stack();
	for(char ch : arr){
		if(stack.isEmpty()){
			stack.push(ch);
		}else{
			char top = (char)stack.lastElement();
			if(ch - top == 1 || ch - top == 2){
				stack.pop();
			}else{
				stack.push(ch);
			}
		}
	}
	if(stack.isEmpty()){
	    return true;
	}
	return false;
}
}

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 7/12/2026, 6:25:36 PM
class Solution:
    def isValid(self, s: str) -> bool:

        hashMap = {'}': '{', ')': '(', ']':'['}
        stack = []

        for ch in s:
            if stack and ch in hashMap and stack[-1] == hashMap[ch]:
                stack.pop()
            else:
                stack.append(ch)

        return not stack
            

        